keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
+// AllocationResultsMaxSize represents the maximum number of
+// entries in allocation.results.
+const AllocationResultsMaxSize = 32
 
-// ResourceHandle holds opaque resource data for processing by a specific kubelet plugin.
-type ResourceHandle struct {
-    // DriverName specifies the name of the resource driver whose kubelet
-    // plugin should be invoked to process this ResourceHandle's data once it
-    // lands on a node. This may differ from the DriverName set in
-    // ResourceClaimStatus this ResourceHandle is embedded in.
+// RequestAllocationResult contains configuration and the allocation result for
+// one request.
+type RequestAllocationResult struct {
+    // DriverName specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
     DriverName string
 
-    // StructuredData captures the result of the allocation for this
-    // particular driver.
-    StructuredData *StructuredResourceHandle
-}
-
-// StructuredResourceHandle is the in-tree representation of the allocation result.
-type StructuredResourceHandle struct {
-    // VendorClassParameters are the per-claim configuration parameters
-    // from the resource class at the time that the claim was allocated.
-    VendorClassParameters runtime.Object
+    // Config contains all the configuration pieces that apply to the request
+    // and that were meant for the driver which handles these devices.
+    // They get collected during the allocation and stored here
+    // to ensure that they remain available while the claim is allocated.
+    //
+    // Entries are list in the same order as in class.config and claim.config,
+    // with class.config entries first.
+    //
+    // +optional
+    Config []DeviceConfiguration
 
-    // VendorClaimParameters are the per-claim configuration parameters
-    // from the resource claim parameters at the time that the claim was
-    // allocated.
-    VendorClaimParameters runtime.Object
+    // RequestName identifies the request in the claim which caused this
+    // device to be allocated. If a request was satisfied by multiple
+    // devices, they are indexed starting with zero in the order
+    // in which they are listed in the results.
+    RequestName string
 
-    // NodeName is the name of the node providing the necessary resources
-    // if the resources are local to a node.
-    NodeName string
+    // This name together with the driver name and the device name field
+    // identify which device was allocated.
+    //
+    // For a driver with node-local devices, it is the name of the node and
+    // the overall ID is `<driver name>/<node name>/<device name>`.
+    //
+    // For other drivers, the overall ID is `<driver name>[/pool
+    // name>]/<device name>`.
```

Comment:
As mentioned above, I think we should onyl have one choice here and that should be `<driver name>/<pool name>/<device name>`. Where `PoolName` *must* be set.

Reply:
+1

Reply:
Updated.